aws-android-sdk-polly/src/test/java/com/amazonaws/services/polly/AmazonPollyPresigningClientTest.java

```diff
@@ -216,4 +213,39 @@ private void assertBasicUriValues(URI uri) {
         Assert.assertEquals(PORT, uri.getPort());
         Assert.assertEquals(RESOURCE_PATH, uri.getPath());
     }
+
+    private static final class QueryParameters {
+        private QueryParameters() {}
+
+        static List<Pair> parse(URI uri) {
+            List<Pair> list = new ArrayList<>();
+            String query = uri.getQuery();
+            if (query == null) {
+                return Collections.emptyList();
+            }
+            for (String pair : query.split("&")) {
+                if (pair != null && pair.length() > 0) {
+                    String[] parts = pair.split("=");
+                    list.add(new Pair(parts[0], parts[1]));
+                }
+            }
+            return list;
+        }
+
+        private static final class Pair {
+            private final android.util.Pair<String, String> androidPair;
```

Comment:
Neat!  I didn't know about `android.util.Pair`.

Reply:
Yea! There are [a few useful things under `android.util.*`](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/androidx-master-dev/core/core/src/main/java/androidx/core/util?source=post_page---------------------------%2F%2F%2F%2F%2F%2F%2F&autodive=0%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F/), and [android.util](https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/util/), it turns out. I haven't looked at many of them.